src/core/server/integration_tests/saved_objects/migrations/zero_downtime/create_index.test.ts

```diff
@@ -0,0 +1,115 @@
+import JSON5 from 'json5';
+import { LogRecord } from '@kbn/logging';
+import { createTestServers, type TestElasticsearchUtils } from '@kbn/core-test-helpers-kbn-server';
+import { getKibanaMigratorTestKit } from '../kibana_migrator_test_kit';
+import { delay } from '../test_utils';
+import { getFooType, getBarType } from './base_types.fixtures';
+
+export const logFilePath = Path.join(__dirname, 'create_index.test.log');
+
+describe('ZDT upgrades - running on a fresh cluster', () => {
+  let esServer: TestElasticsearchUtils['es'];
+
+  const startElasticsearch = async () => {
+    const { startES } = createTestServers({
+      adjustTimeout: (t: number) => jest.setTimeout(t),
+      settings: {
+        es: {
+          license: 'basic',
+        },
+      },
+    });
+    return await startES();
+  };
+
+  beforeAll(async () => {
+    await fs.unlink(logFilePath).catch(() => {});
+    esServer = await startElasticsearch();
+  });
+
+  afterAll(async () => {
+    await esServer?.stop();
+    await delay(10);
+  });
+
+  it('create the index with the correct mappings and meta', async () => {
+    const fooType = getFooType();
+    const barType = getBarType();
+
+    const { runMigrations, client } = await getKibanaMigratorTestKit({
+      kibanaIndex: '.kibana',
+      kibanaVersion: '8.7.0',
+      logFilePath,
+      types: [fooType, barType],
+      settings: {
+        migrations: {
+          algorithm: 'zdt',
+        },
+      },
+    });
+
+    const result = await runMigrations();
+
+    expect(result).toEqual([
+      {
+        destIndex: '.kibana',
+        elapsedMs: expect.any(Number),
+        status: 'patched',
+      },
+    ]);
+
+    const indices = await client.indices.get({ index: '.kibana*' });
+
+    expect(Object.keys(indices)).toEqual(['.kibana_1']);
+
+    const index = indices['.kibana_1'];
+    const aliases = Object.keys(index.aliases ?? {}).sort();
+    const mappings = index.mappings ?? {};
+    const mappingMeta = mappings._meta ?? {};
+
+    expect(aliases).toEqual(['.kibana', '.kibana_8.7.0']);
+
+    expect(mappings.properties).toEqual(
+      expect.objectContaining({
+        foo: fooType.mappings,
+        bar: barType.mappings,
+      })
+    );
+
+    expect(mappingMeta).toEqual({
+      docVersions: {
+        foo: 2,
+        bar: 1,
+      },
+      mappingVersions: {
+        foo: 2,
+        bar: 1,
+      },
+    });
+
+    const logFileContent = await fs.readFile(logFilePath, 'utf-8');
+    const records = logFileContent
+      .split('\n')
+      .filter(Boolean)
+      .map((str) => JSON5.parse(str)) as LogRecord[];
+
+    const expectLogsContains = (messagePrefix: string) => {
+      expect(records.find((entry) => entry.message.includes(messagePrefix))).toBeDefined();
+    };
+
+    expectLogsContains('INIT -> CREATE_TARGET_INDEX');
+    expectLogsContains('CREATE_TARGET_INDEX -> UPDATE_ALIASES');
+    expectLogsContains('UPDATE_ALIASES -> DONE');
+    expectLogsContains('Migration completed');
```

Comment:
Minor comment: I've seen this approached a couple different ways (partly my fault). I think it is worth consolidating:

1. `JSON5.parse` way (used here)
2. Reading full string content from log file then `expect(logs).toMatch('A -> B')`

Way (2) is provided by the `kibana_migrator_test_kit` and used in `src/core/server/integration_tests/saved_objects/migrations/group3/skip_reindex.test.ts`, for example.

Not major, but for these Jest integration tests would be nice to use one approach. Personally I prefer (2) bc it is "good enough" and avoids using a lib. Disclaimer: I have not benchmarked either of these.

Reply:
cc @gsoldevila

Reply:
In the recent enhancements to the migration logic I ended up using (2), parsing the whole text into a string, and then using `expect(logs).toMatch('certain string');`.

I tried to simplify this and extract it into a method that allowed to check multiple strings were present in the logs. Whilst it was clearer in terms of readability, it added more calls in the stack, and in case of failure it prevented me from seeing the real call that was failing (Jest is currently printing 3 lines of the stack). Thus, I tend to agree with JL for these 2 reasons (avoid using lib + clearer stack traces).

Reply:
>  Personally I prefer (2) bc it is "good enough" 

Personal opinion, but I don't think it's good enough for all scenarios.

The limitations of the `expect(myWholeFileAsAString).toContains(someText)` is that you won't be able to perform advanced tests on the order of the logs (and ihmo we *should* assert against the order of the logs too).

Like what I did here a while ago:

https://github.com/elastic/kibana/blob/fcf10981e5633662e5748199ae10be984f85fa48/src/core/server/integration_tests/saved_objects/migrations/group2/collects_corrupt_docs.test.ts#L122-L127

Reply:
> I don't think it's good enough for all scenarios...the order of the logs

Good point! I hadn't thought of order. Just brainstorming here:

```ts
  expectLogOrder(logs, stringToMatch: string[])
  ...
  const logs = (await readLog()).split('\n')
  expectLogOrder(logs, ['INIT -> FOO', 'FOO -> BAR', ...]) // or other messages
```

But yeah, happy to go with whoever has the strongest opinion. Would just be nice to keep the test itself super lean.